adalflow/adalflow/components/model_client/bedrock_client.py

```diff
@@ -0,0 +1,154 @@
+"""AWS Bedrock ModelClient integration."""
+
+import os
+from typing import Dict, Optional, Any, Callable
+import backoff
+import logging
+
+from adalflow.core.model_client import ModelClient
+from adalflow.core.types import ModelType, CompletionUsage, GeneratorOutput
+
+import boto3
+from botocore.config import Config
+
+log = logging.getLogger(__name__)
+
+bedrock_runtime_exceptions = boto3.client(
+    service_name="bedrock-runtime",
+    region_name=os.getenv("AWS_REGION_NAME", "us-east-1")
+).exceptions
+
+
+def get_first_message_content(completion: Dict) -> str:
+    r"""When we only need the content of the first message.
+    It is the default parser for chat completion."""
+    return completion['output']['message']['content'][0]['text']
+
+
+__all__ = ["BedrockAPIClient", "get_first_message_content", "bedrock_runtime_exceptions"]
+
+
+class BedrockAPIClient(ModelClient):
+    __doc__ = r"""A component wrapper for the Bedrock API client.
+    Visit https://docs.aws.amazon.com/bedrock/latest/APIReference/welcome.html for more api details.
+    """
+
+    def __init__(
+            self,
+            aws_profile_name=None,
```

Comment:
please add type hinting

Reply:
Also need more detailed instruction on how to set up the env parameters. For now, I dont know what value i need to pass to `aws_connection_timeout=None,
            aws_read_timeout=None,`
            
 And why do they matter?
 
 Tokens and profiles are easy to understand